### PR TITLE
Add glob

### PR DESCRIPTION
**Pull request by katcipis**
Just that, and some minor refactorings

### CODE REVIEW

**Comment by vitorarins** [comment]
This closes #183.

**Comment by katcipis** [comment]
@tiago4orion @lborguetti @vitorarins Can we merge this ?

File: internal/sh/builtin/glob.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+func (p *globFn) ArgNames() []string {
+	return []string{"fmt", "args..."}
+}
+
+func (g *globFn) Run(in io.Reader, out io.Writer, e io.Writer) ([]sh.Obj, error) {
+	listobjs := []sh.Obj{}
+	matches, err := filepath.Glob(g.pattern)
+	if err != nil {
+		return nil, fmt.Errorf("glob:error: %q", err)
```
Comment (tiago4orion) [inline internal/sh/builtin/glob.go]: using `fmt.Errorf` instead of `errors.NewError`.

Comment (katcipis) [inline internal/sh/builtin/glob.go]: Sorry, I was not aware of errors.NewError

Comment (katcipis) [inline internal/sh/builtin/glob.go]: fixed

**Pull request merged**